tools/clang-tidy-checks/15.x-patches/0001-Max-tokens-checks.diff

```diff
@@ -0,0 +1,309 @@
++                            unsigned MaxTokenCount, bool IsFromPragma) override {
++    MaxTokenCount = MaxTokenCount == 0 ? Check->FMaxTokens : MaxTokenCount;
++    diag(Loc, TokenCount, MaxTokenCount, IsFromPragma);
++  }
++
++private:
++  MaxTokensCheck *Check;
++};
++} // namespace
++
++void MaxTokensCheck::registerPPCallbacks(const SourceManager &SM,
++                                         Preprocessor *PP,
++                                         Preprocessor *ModuleExpanderPP) {
++  PP->addPPCallbacks(std::make_unique<MaxTokensPPCallbacks>(this));
++}
++
++void MaxTokensCheck::storeOptions(ClangTidyOptions::OptionMap &Opts) {
++  Options.store(Opts, "FMaxTokens", FMaxTokens);
++}
++
++} // namespace misc
++} // namespace tidy
++} // namespace clang
+diff --git a/clang-tools-extra/clang-tidy/misc/MaxTokensCheck.h b/clang-tools-extra/clang-tidy/misc/MaxTokensCheck.h
+new file mode 100644
+index 000000000..5e73923ad
+--- /dev/null
++++ b/clang-tools-extra/clang-tidy/misc/MaxTokensCheck.h
+@@ -0,0 +1,37 @@
++//===--- MaxTokensCheck.h - clang-tidy --------------------------*- C++ -*-===//
++//
++// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
++// See https://llvm.org/LICENSE.txt for license information.
++// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
++//
++//===----------------------------------------------------------------------===//
++
++#ifndef LLVM_CLANG_TOOLS_EXTRA_CLANG_TIDY_MISC_MAXTOKENSCHECK_H
++#define LLVM_CLANG_TOOLS_EXTRA_CLANG_TIDY_MISC_MAXTOKENSCHECK_H
++
++#include "../ClangTidyCheck.h"
++
++namespace clang {
++namespace tidy {
++namespace misc {
++
++/// Implements the -Wmax-tokens clang flag as a check
++///
++/// For the user-facing documentation see:
++/// http://clang.llvm.org/extra/clang-tidy/checks/misc-max-tokens.html
++class MaxTokensCheck : public ClangTidyCheck {
++public:
++  MaxTokensCheck(StringRef Name, ClangTidyContext *Context)
++      : ClangTidyCheck(Name, Context),
++        FMaxTokens(Options.get("FMaxTokens", 0)) {}
++  const unsigned FMaxTokens;
++  void registerPPCallbacks(const SourceManager &SM, Preprocessor *PP,
++                           Preprocessor *ModuleExpanderPP) override;
++  void storeOptions(ClangTidyOptions::OptionMap &Opts) override;
++};
++
++} // namespace misc
++} // namespace tidy
++} // namespace clang
++
++#endif // LLVM_CLANG_TOOLS_EXTRA_CLANG_TIDY_MISC_MAXTOKENSCHECK_H
+diff --git a/clang-tools-extra/clang-tidy/misc/MiscTidyModule.cpp b/clang-tools-extra/clang-tidy/misc/MiscTidyModule.cpp
+index 127889230..29c576ea7 100644
+--- a/clang-tools-extra/clang-tidy/misc/MiscTidyModule.cpp
++++ b/clang-tools-extra/clang-tidy/misc/MiscTidyModule.cpp
+@@ -41,6 +41,8 @@ public:
+         "misc-const-correctness");
+     CheckFactories.registerCheck<DefinitionsInHeadersCheck>(
+         "misc-definitions-in-headers");
++    CheckFactories.registerCheck<MaxTokensCheck>(
```

Comment:
I think you need to include the header here.

Reply:
Indeed, not sure why it didn't generate two conflicts when I was doing rebase